Source/WebKit/WebProcess/cocoa/HandleXPCEndpointMessages.mm

```diff
@@ -48,6 +57,21 @@ void handleXPCEndpointMessages(xpc_object_t event, const char* messageName)
         return;
     }
 #endif
+
+#if USE(EXTENSIONKIT)
+    if (!strcmp(messageName, layerHostingContextChangedMessage)) {
+        callOnMainRunLoop([event = OSObjectPtr<xpc_object_t>(event)] () {
+            auto identifier = xpc_dictionary_get_uint64(event.get(), mediaplayerIdentifierKey);
+            auto width = xpc_dictionary_get_uint64(event.get(), presentationSizeWidthKey);
+            auto height = xpc_dictionary_get_uint64(event.get(), presentationSizeHeightKey);
```

Comment:
It seems like you could avoid the `width` and `height` lookups if the `player` weak pointer happens to be nullptr.

Reply:
Fixed!